src/main/cpp/drivers/adapter.cpp

```diff
@@ -27,97 +27,130 @@
 namespace MWEngine {
 namespace DriverAdapter {
 
-#if DRIVER == 0
-    OPENSL_STREAM* driver_openSL = nullptr; // OpenSL
-#elif DRIVER == 1
-    AAudio_IO* driver_aAudio = nullptr;     // AAudio
-#endif
-
-    bool create() {
-
-#if DRIVER == 0
-
-        Debug::log( "DriverAdapter::initializing OpenSL driver");
-
-        // OpenSL
-        driver_openSL = android_OpenAudioDevice(
-            AudioEngineProps::SAMPLE_RATE, AudioEngineProps::INPUT_CHANNELS,
-            AudioEngineProps::OUTPUT_CHANNELS, AudioEngineProps::BUFFER_SIZE
-        );
-        return ( driver_openSL != NULL );
-
-#elif DRIVER == 1
+    Drivers::types _driver       = Drivers::types::OPENSL;
 
-        Debug::log( "DriverAdapter::initializing AAudio driver");
-
-        // AAudio
-        driver_aAudio = new AAudio_IO(
-            AudioEngineProps::INPUT_CHANNELS, AudioEngineProps::OUTPUT_CHANNELS
-        );
-
-        if ( driver_aAudio == nullptr ) {
-            return false;
-        }
-        // TODO: allow specifying these from the outside?
-        // driver_aAudio->setDeviceId();
-        // driver_aAudio->setRecordingDeviceId();
-        driver_aAudio->setBufferSizeInBursts( 1 ); // Google provides {0, 1, 2, 4, 8} as values
-
-        return true;
+    AAudio_IO* driver_aAudio     = nullptr;
+    OPENSL_STREAM* driver_openSL = nullptr;
+#ifdef MOCK_ENGINE
+    Mock_IO*       driver_mocked = nullptr;
 #endif
 
+    MWEngine_renderCallback render      = nullptr;
+    MWEngine_writeCallback  writeOutput = nullptr;
+    MWEngine_readCallback   getInput    = nullptr;
+
```

Comment:
Avoid creating large groups of conditionals statements by defining the driver-specific implementation of these functions within the _create()_-function.